lib/ansible/cli/galaxy.py

```diff
@@ -1345,15 +1348,17 @@ def _execute_install_collection(
         no_deps = context.CLIARGS['no_deps']
         force_with_deps = context.CLIARGS['force_with_deps']
         disable_gpg_verify = context.CLIARGS['disable_gpg_verify']
+        collection_path_warnings = context.CLIARGS['collection_path_warnings']
         # If `ansible-galaxy install` is used, collection-only options aren't available to the user and won't be in context.CLIARGS
         allow_pre_release = context.CLIARGS.get('allow_pre_release', False)
         upgrade = context.CLIARGS.get('upgrade', False)
 
-        collections_path = C.COLLECTIONS_PATHS
-        if len([p for p in collections_path if p.startswith(path)]) == 0:
-            display.warning("The specified collections path '%s' is not part of the configured Ansible "
-                            "collections paths '%s'. The installed collection will not be picked up in an Ansible "
-                            "run, unless within a playbook-adjacent collections directory." % (to_text(path), to_text(":".join(collections_path))))
+        if collection_path_warnings:
+            collections_path = C.COLLECTIONS_PATHS
+            if len([p for p in collections_path if p.startswith(path)]) == 0:
```

Comment:
You could extend this condition and avoid having to indent a bunch of code:
```suggestion
            display_path_warning = collection_path_warnings and len([p for p in collections_path if p.startswith(path)])
            if display_path_warning:
```

Reply:
Indeed. That's probably better.